NitroxPatcher/Patches/Dynamic/SubConsoleCommand_OnConsoleCommand_sub_Patch.cs

```diff
@@ -1,19 +1,63 @@
+using System.Collections.Generic;
 using System.Reflection;
+using System.Reflection.Emit;
+using HarmonyLib;
+using NitroxClient.GameLogic;
+using NitroxModel.DataStructures.GameLogic;
 using NitroxModel.Helper;
+using UnityEngine;
 
 namespace NitroxPatcher.Patches.Dynamic;
 
 /// <summary>
-/// Called whenever a Cyclops or Seamoth is spawned. Nitrox already has its own command to spawn vehicles.
-/// This patch is only meant to block the method from executing, causing two vehicles to be spawned instead of one
+/// Prevents local player from using "sub" command without at least the <see cref="Perms.MODERATOR"/> permissions.
+/// Once they have the permissions, sync this command.
 /// </summary>
 public sealed partial class SubConsoleCommand_OnConsoleCommand_sub_Patch : NitroxPatch, IDynamicPatch
 {
-    private static readonly MethodInfo TARGET_METHOD = Reflect.Method((SubConsoleCommand t) => t.OnConsoleCommand_sub(default));
+    internal static readonly MethodInfo TARGET_METHOD = Reflect.Method((SubConsoleCommand t) => t.OnConsoleCommand_sub(default));
 
-    public static bool Prefix()
+    public static bool Prefix(NotificationCenter.Notification n)
     {
-        Log.InGame(Language.main.Get("Nitrox_CommandNotAvailable"));
-        return false;
+        if (Resolve<LocalPlayer>().Permissions < Perms.MODERATOR)
+        {
+            Log.InGame(Language.main.Get("Nitrox_MissingPermission").Replace("{PERMISSION}", Perms.MODERATOR.ToString()));
+            return false;
+        }
+
+        string text = (string)n.data[0];
+        if (!string.IsNullOrEmpty(text) && !text.ToLowerInvariant().Equals("cyclops"))
+        {
+            Log.InGame(Language.main.Get("Nitrox_CommandNotAvailable"));
+            return false;
+        }
+        return true;
+    }
+
+    /*
+     * REPLACE:
+     * LightmappedPrefabs.main.RequestScenePrefab(text, new LightmappedPrefabs.OnPrefabLoaded(this.OnSubPrefabLoaded));
+     * BY:
+     * LightmappedPrefabs.main.RequestScenePrefab(text, new LightmappedPrefabs.OnPrefabLoaded(SubConsoleCommand_OnConsoleCommand_sub_Patch.WrappedCallback));
+     */
+    public static IEnumerable<CodeInstruction> Transpiler(IEnumerable<CodeInstruction> instructions)
+    {
+        return new CodeMatcher(instructions).MatchEndForward([
+                                                new CodeMatch(OpCodes.Stfld),
+                                                new CodeMatch(OpCodes.Ldsfld),
+                                                new CodeMatch(OpCodes.Ldloc_0),
+                                                new CodeMatch(OpCodes.Ldarg_0)
+                                            ])
+                                            .SetOpcodeAndAdvance(OpCodes.Ldnull)
```

Comment:
I am guessing that you replace the `Ldarg_0` here to null as it's no longer a local method being called. Would a `nop` also work? I am guessing that `nop` is more clear and can't produce any side effects.

Reply:
Results of the discussion in the Discord:
"newobj constructor takes an object and a native int, the object being the instance (null in our case for  static) and the native int being the pointer given by ldftn.
ldftn will in either way (static or instance method) just give a pointer without the instance mentionned. It's about the instance we give to the constructor"